### PR TITLE
Refund child in seperate contract

### PR DESCRIPTION
**Pull request by CJentzsch**

### CODE REVIEW

**Comment by chriseth** [comment]
The withdraw function should have a guard so that it cannot be called before the hard fork block number.

File: refund.sol [hunk 1]
```diff
@@ -17,96 +17,22 @@ along with the DAO.  If not, see <http://www.gnu.org/licenses/>.
...
-        withdraw(_child);
-    }
-
-    function withdraw(){
-        withdraw(mother);
-    }
+    function withdraw(DAO _dao) {
+        if (block.number < 0xfffff) throw; // to be replaced by HF block number
+        if (!refundChild) {
+            refundChild = true;
```
Comment (chriseth) [inline refund.sol]: I don't think there is a need for the flag, we can use the address for both.

File: refund.sol [hunk 2]
```diff
@@ -17,96 +17,22 @@ along with the DAO.  If not, see <http://www.gnu.org/licenses/>.
...
+            childRefund = new ChildRefund()
+            if (!childRefund.send(174558484783966581114679) || address(childRefund) == 0) throw;
```
Comment (chriseth) [inline refund.sol]: Is the number fixed? Is it "initial dao - dark dao - white dao"?

Comment (CJentzsch) [inline refund.sol]: yes, basically all ETH that is in all childDAO, minus the attacker (white and black) daos.

File: refund.sol [hunk 3]
```diff
@@ -17,96 +17,20 @@ along with the DAO.  If not, see <http://www.gnu.org/licenses/>.
...
-        withdraw(_child);
-    }
-
-    function withdraw(){
-        withdraw(mother);
-    }
+    function withdraw(DAO _dao) {
+        if (block.number < 0xfffff) throw; // to be replaced by HF block number
+        if (address(childRefund) == 0)) {
+            childRefund = new ChildRefund()
```
Comment (chriseth) [inline refund.sol]: Missing `;`

Comment (CJentzsch) [inline refund.sol]: fixed

**Comment by Sulfurtank** [comment]
bur ether and dont refund anyone, do the right thing scumbags

**Comment by Sulfurtank** [comment]
burn eth and dont refund anyone do the right thing scumbags

**Pull request closed without merging**